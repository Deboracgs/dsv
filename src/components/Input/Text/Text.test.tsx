import { fireEvent, render } from "@testing-library/react";
import WrapperMock from "../../../utils/wrapperMock";
import InputText, { InputTextProps } from "./Text";

describe("Input Text Component", () => {
  const defaultProps: InputTextProps = {
    id: "unit-text", 
    label: "label", 
    defaultValue: 0,
    onChange: jest.fn()
  };

  const rendered = (props: InputTextProps) =>
    render(
      <WrapperMock>
        <InputText {...props} />
      </WrapperMock>
    );

  it("render", () => {
    const onChangeSpy = jest.spyOn(defaultProps, "onChange");
    const { asFragment, container } = rendered(defaultProps);

    const element = container.querySelector("#unit-text") as HTMLInputElement;

    expect(element).toBeDefined();
    expect(element.defaultValue).toBe("0");
    fireEvent.change(element, {target: {value: 3}});
    
    expect(onChangeSpy).toHaveBeenCalled();
    expect(asFragment()).toMatchSnapshot();
  });

});
