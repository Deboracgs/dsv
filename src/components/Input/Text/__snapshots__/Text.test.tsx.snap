// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Text Component render 1`] = `
<DocumentFragment>
  <div
    data-testid="input-text-unit-text"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      data-testid="input-text-unit-text"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="unit-text"
        id="unit-text-label"
      >
        label
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="unit-text"
          name="input-text-unit-text"
          type="text"
          value="0"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1in441m"
          >
            <span>
              label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
